packaging/msi/InstallFolderComponent.wxs

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs"
+     xmlns:ui="http://wixtoolset.org/schemas/v4/wxs/ui"
+     xmlns:util="http://wixtoolset.org/schemas/v4/wxs/util">
+    <Fragment>
+        <!-- Define the directory structure -->
+        <StandardDirectory Id="ProgramFiles64Folder">
+            <Directory Id="INSTALLFOLDER" Name="ZeroC\Ice-$(Version)">
+                <Directory Id="BinFolder" Name="bin" />
+                <Directory Id="SliceFolder" Name="slice" />
+                <Directory Id="ConfigFolder" Name="config" />
+            </Directory>
+        </StandardDirectory>
+        <StandardDirectory Id="ProgramMenuFolder">
+            <Directory Id="IceStartMenu" Name="Ice $(Version)" />
+        </StandardDirectory>
+
+        <Component Id="AddToPath" Guid="361FB9AF-3452-45F8-93E9-AAA1E97CB1AA" Condition="ADD_TO_PATH = 1">
+            <Environment
+                Id="AddToSystemPath"
+                Name="PATH"
+                Action="set"
+                Part="last"
+                System="yes"
+                Value="[BinFolder]"
+                Permanent="no" />
+        </Component>
```

Comment:
This Environment element wrap in a conditional component, adds BinFolder to path based on the ADD_TO_PATH property set from the UI checkbox.